### PR TITLE
Document alternative process to setup a custom hostname for a Service

### PR DESCRIPTION
**Pull request by jonatasbaldin**
Adds an alternative method to fix [Serving #1185](https://github.com/knative/serving/issues/1185) and #1964.

## Proposed Changes 
- Adds an alternative documentation to create custom Service FQDN.

### CODE REVIEW

File: docs/serving/using-a-custom-domain-per-service.md [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+       {{end -}}
+       .{{.Domain}}
+   ```
+
+   Save and close your editor.
+
+## Edit the Service
+
+1. In a Service definition, add the `custom-hostname` annotation:
+
```
Comment (mattmoor) [inline docs/serving/using-a-custom-domain-per-service.md:42]: Remove trailing whitespace:
```suggestion

```

File: docs/serving/using-a-custom-domain-per-service.md [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+## Verify the changes
+
+1. Verify that the Service was created with the specified hostname:
+
+  ```shell
+  kubectl get ksvc hello-world
+
+  NAME          URL                              LATESTCREATED       LATESTREADY         READY   REASON
+  hello-world   http://hello-world.example.com   hello-world-nfqh2   hello-world-nfqh2   True
```
Comment (mattmoor) [inline docs/serving/using-a-custom-domain-per-service.md:65]: Remove trailing whitespace:
```suggestion
  hello-world   http://hello-world.example.com   hello-world-nfqh2   hello-world-nfqh2   True
```

File: docs/serving/using-a-custom-domain-per-service.md [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+## Verify the changes
+
+1. Verify that the Service was created with the specified hostname:
+
+  ```shell
+  kubectl get ksvc hello-world
+
+  NAME          URL                              LATESTCREATED       LATESTREADY         READY   REASON
+  hello-world   http://hello-world.example.com   hello-world-nfqh2   hello-world-nfqh2   True    
+  ```
```
Comment (mattmoor) [inline docs/serving/using-a-custom-domain-per-service.md:66]: Add trailing newline:
```suggestion
  ```

```

**Comment by jonatasbaldin** [comment]
hey @bsideup, could u take a look?

File: docs/serving/using-a-custom-domain-per-service.md [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+
+## Edit using kubectl
+
+1. Edit the `domainTemplate` entry on the `config-network` configuration:
+
+   ```shell
+   kubectl edit cm config-network --namespace knative-serving
+   ```
+
+   Replace the `domainTemplate` with the following (the spaces must be respected):
```
Comment (bsideup) [inline docs/serving/using-a-custom-domain-per-service.md:22]: Consider adding a link to https://github.com/knative/serving/blob/3e920bd986ecf6d22d506a273a8ed15c4f266eec/config/core/configmaps/network.yaml#L69 for a wider explanation of available options

**Comment by bsideup** [comment]
@jonatasbaldin looks nice! Added a minor note about the underlying sorcery, but overall LGTM 👍 

P.S. thanks for the mention of the article 😊

File: docs/serving/using-a-custom-domain-per-service.md (hunk 4 above)
Comment (jonatasbaldin) [reply docs/serving/using-a-custom-domain-per-service.md:22]: Will do. Will use the master branch, tho :)

**Pull request merged**